### PR TITLE
feat(admin-dashboard): created recent orders functionality

### PR DESCRIPTION
**Pull request by rohansen856**
## Overview

1. This PR fixes or fixes part of #89 .
2. This PR does the following: created recent orders details card with data fetching.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [x] The PR is made from a branch that's **not** called "main/master".

## Proof that changes are correct
Screenshot 2024-06-16 232750
Screenshot 2024-06-16 232726

responsiveness:
Screenshot 2024-06-16 232508
Screenshot 2024-06-16 232534
Screenshot 2024-06-16 232523

works done in this pr:
### frontend
1. created a list view of the recent orders (orders done inn the same day).
2. created order details card, viewable when clicked `View Details`

### backend
1. modified the `orders.service.ts` file to add the logic to get the data in required format
2. modified the `orders.controller.ts` file to add the route handler to get the details.
3. fixed the seed file to push the data in better format.

## PR Pointers

- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL".
- Never force push. If you do, your PR will be closed.

### CODE REVIEW

**Comment by RISHIKESHk07** [comment]
@rohansen856 we will be having relationship between tickets and orders as well , it make it easier for the other issue for data  fetching , please make changes according to this schema 
WhatsApp Image 2024-06-17 at 9 46 09 AM

**Comment by rohansen856** [comment]
@RISHIKESHk07 i will fix the issue and update the schema with the correct relationships.

**Comment by rohansen856** [comment]
@RISHIKESHk07  @Pratik2026  i have updated the `schema.prisma` file to make relations between orders, tickets and schedule schma. also updated the data fetching method in `order.service.ts` in my latest pr.
the current relationship is:
```prisma
model Schedule {
    id            String   @id
    busNumber     String
    bus           Bus      @relation(fields: [busNumber], references: [number])
    checkpoints   String[]
    from          String
    to            String
    departureTime String
    days          String[]
    ticketPrice   Int
    orders        Order[]
}

model Ticket {
    id             Int @id @default(autoincrement())
    orderId        String @unique
    passengerEmail String[]
    order          Order @relation(fields: [orderId], references: [id])
}

model Order {
    id         String     @id
    userId     String
    scheduleId String
    status     String     @default("pending") // how to make this enum
    amount     Int
    attempts   Int        @default(0)
    receipt    String     @unique
    createdAt  DateTime   @default(now())
    user       Users      @relation(fields: [userId], references: [id])
    ticket     Ticket?
    schedule   Schedule  @relation(fields: [scheduleId], references: [id])
}
```


 thhank you.

**Review by RISHIKESHk07 — APPROVED** [review]

**Comment by RISHIKESHk07** [comment]
@Pratik2026 LGTM  codewise , any changes as per you ?

File: frontend/src/components/AdminPanel/OrderDetailsCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
+import { Typography, styled, Box, useTheme } from '@mui/material';
+import  Clear  from "@mui/icons-material/clear"
```
Comment (RISHIKESHk07) [inline frontend/src/components/AdminPanel/OrderDetailsCard/index.tsx:2]: @rohansen856  double check this  line @mui/icons-material/clear"

**Review by RISHIKESHk07 — CHANGES_REQUESTED** [review]
@rohansen856 did you verify if page works when we add a ticket through the razorpay UI (dummy order) instead of seeding the data ? i tried it here (Left: Busify & Right:Ptisma Studio) , there is a ticket created but does not appear in the admin page , maybe its a date issue , you can test razorpay UPI with success@razorpay
image

File: frontend/src/components/AdminPanel/OrderDetailsCard/index.tsx (hunk 1 above)
Comment (rohansen856) [reply frontend/src/components/AdminPanel/OrderDetailsCard/index.tsx:2]: I will fix the issue with the formatting and add the semicolon(`;`) in this line.

**Comment by rohansen856** [comment]
> @rohansen856 did you verify if page works when we add a ticket through the razorpay UI (dummy order) instead of seeding the data ? i tried it here (Left: Busify & Right:Ptisma Studio) , there is a ticket created but does not appear in the admin page , maybe its a date issue , you can test razorpay UPI with success@razorpay image

@RISHIKESHk07 i will double check the payments issue with razorpay, and also check any issues with the data.

**Comment by rohansen856** [comment]
@RISHIKESHk07 i have solved the following issues in the latest pr:
1. updated the prisma `Ticket` schema with the discussed changes.
2. updated the order of database seeding in `prisma/seed.mjs`.
3. updated `orders.service.ts` to fix the issue with the date object in razorpay payment datetime.
4. fixed the frontend issue with parsing the datetime by encapsulation the field in `new Date()`.
5. fixed the issue with formatting in `frontend/src/components/AdminPanel/OrderDetailsCard/index.tsx`.
proof of work:

https://github.com/bsoc-bitbyte/busify/assets/108976786/e01cbc5c-5195-46ae-b71b-cd60db4765d8


https://github.com/bsoc-bitbyte/busify/assets/108976786/57c85be1-8913-4363-9193-5273edd08616

thank you.

File: frontend/src/components/AdminPanel/OrderDetailsCard/index.tsx (hunk 1 above)
Comment (RISHIKESHk07) [reply frontend/src/components/AdminPanel/OrderDetailsCard/index.tsx]: @rohansen856 nah the probelm is the its "@mui/icons-material/Clear" , on my end i see it giving a error for ...../clear

**Review by RISHIKESHk07 — APPROVED** [review]

**Pull request merged**